vortex-array/src/patches.rs

```diff
@@ -284,7 +295,18 @@ impl Patches {
     }
 
     pub fn take_map(&self, take_indices: &ArrayData) -> VortexResult<Option<Self>> {
-        let take_indices = take_indices.clone().into_primitive()?;
+        let take_indices = try_cast(
+            take_indices,
```

Comment:
What is this cast for?

Reply:
Signed integers are, apparently, allowed as take indices. `take_map_helper` relies on the new `NativeIndexPType` which I only implemented for the four unsigned types. This type lets me avoid converting an indices slice with, for example, the type `&[u8]` into a `&[usize]`. Instead, I iterate directly on the u8s converting them to usize only when I'm using them to index an array or to compare to other indices.

We could benchmark, but I was pretty convinced by the downscale integer changes that inflating arrays of small integers into big integers is fairly expensive.

Reply:
You could change the `match_each_unsigned_integer_ptype` to `match_each_integer_ptype` and remove this cast.

Reply:
Done.